src/Framework.Cqrs.Extensions.AspNetCore/AspNetCoreResolverBuilder.cs

```diff
@@ -0,0 +1,43 @@
+﻿namespace PetProjects.Framework.Cqrs.Extensions.AspNetCore
+{
+    using Microsoft.Extensions.DependencyInjection;
+    using PetProjects.Framework.Cqrs.Commands;
+    using PetProjects.Framework.Cqrs.DependencyResolver;
+    using PetProjects.Framework.Cqrs.Queries;
+
+    internal class AspNetCoreResolverBuilder : IDependencyResolverBuilder
+    {
+        private readonly IServiceCollection collection;
+
+        public AspNetCoreResolverBuilder(IServiceCollection collection)
+        {
+            this.collection = collection;
+        }
+
+        public void RegisterQueryHandlerAsync<THandler, TQuery, TResponse>()
+            where THandler : class, IQueryHandlerAsync<TQuery, TResponse>
+            where TQuery : IQuery<TResponse>
+        {
+            this.collection.AddScoped<IQueryHandlerAsync<TQuery, TResponse>, THandler>();
```

Comment:
With scoped I suppose and hope :) we won't have database access inside these handlers. 

Am I right?

Aren't we removing flexibility to the user, by giving a default scoped behavior?

We are hiding "behavior" which in my opinion is bad. We should let the user decide what kind of lifetime he wants for the handlers.

At least we should show in the method name that it has a scoped lifetime.

Reply:
I know we give him the option to create his own resolver, but I think we hiding some details.

Reply:
>With scoped I suppose and hope :) we won't have database access inside these handlers.
>
> Am I right?

What do you mean? What does the lifetime have to do with database access?


>Aren't we removing flexibility to the user, by giving a default scoped behavior?
>
>We are hiding "behavior" which in my opinion is bad. We should let the user decide what kind of >lifetime he wants for the handlers.
>
>At least we should show in the method name that it has a scoped lifetime.

This is an internal class, it will only be used by this framework in the aspnetcore extension method. But I agree that it would be an interesting addition to let the user specify the lifetime of both the dependency resolver and the handlers. Added in 1a985efc9e632d2ef0fcd91b7da1d44f9d3d8541

Reply:
When I meant access to the Database is creating multiple objects of Db instantiation. Which makes no sense since it will be injected in the ctor of the class. This was a stupid idea that passed through my mind.